cluster-autoscaler/cloudprovider/gce/cache.go

```diff
@@ -195,10 +201,11 @@ func (gc *GceCache) GetMigForInstance(instanceRef GceRef) (Mig, error) {
 	return nil, nil
 }
 
-func (gc *GceCache) removeInstancesForMig(migRef GceRef) {
+func (gc *GceCache) removeInstancesForMigs(migRef GceRef) {
 	for instanceRef, instanceMigRef := range gc.instanceRefToMigRef {
 		if migRef == instanceMigRef {
 			delete(gc.instanceRefToMigRef, instanceRef)
+			delete(gc.instancesFromUnknownMigs, instanceRef)
```

Comment:
It should not be needed. No harm though.

Reply:
I think we should clear the Unknown map in RegenerateInstancesCache.
Without it we have (probably harmless leak) as removeInstancesForMigs will not be called for 
Migs which no longer exist